nilearn/decoding/tests/test_space_net.py

```diff
@@ -0,0 +1,347 @@
+import os
+import warnings
+import itertools
+from functools import partial
+from nose import SkipTest
+from nose.tools import (assert_equal, assert_true, assert_false,
+                        assert_raises)
+import numpy as np
+import nibabel
+from sklearn.datasets import load_iris
+from sklearn.utils import extmath
+from sklearn.linear_model import Lasso
+from sklearn.utils import check_random_state
+from sklearn.linear_model import LogisticRegression
+from nilearn.decoding.space_net import (
+    _EarlyStoppingCallback, _space_net_alpha_grid, MNI152_BRAIN_VOLUME,
+    path_scores, BaseSpaceNet, _crop_mask, _univariate_feature_screening,
+    _get_mask_volume, SpaceNetClassifier, SpaceNetRegressor)
+from nilearn.decoding.space_net_solvers import (_graph_net_logistic,
+                                                _graph_net_squared_loss)
+
+mni152_brain_mask = (
+    "/usr/share/fsl/data/standard/MNI152_T1_1mm_brain_mask.nii.gz")
+logistic_path_scores = partial(path_scores, is_classif=True)
+squared_loss_path_scores = partial(path_scores, is_classif=False)
+
+# Data used in almost all tests
+from .test_same_api import to_niimgs
+size = 4
+from .simulate_graph_net_data import create_graph_net_simulation_data
+X_, y, w, mask = create_graph_net_simulation_data(
+    snr=1., n_samples=10, size=size, n_points=5, random_state=42)
+X, mask = to_niimgs(X_, [size] * 3)
+
+
+def test_space_net_alpha_grid(n_samples=4, n_features=3):
+    rng = check_random_state(42)
+    X = rng.randn(n_samples, n_features)
+    y = np.arange(n_samples)
+
+    for l1_ratio, is_classif in itertools.product([.5, 1.], [True, False]):
+        alpha_max = np.max(np.abs(np.dot(X.T, y))) / l1_ratio
+        np.testing.assert_almost_equal(_space_net_alpha_grid(
+                X, y, n_alphas=1, l1_ratio=l1_ratio,
+                logistic=is_classif), alpha_max)
+
+    for l1_ratio, is_classif in itertools.product([.5, 1.], [True, False]):
+        alpha_max = np.max(np.abs(np.dot(X.T, y))) / l1_ratio
+        for n_alphas in range(1, 10):
+            alphas = _space_net_alpha_grid(
+                X, y, n_alphas=n_alphas, l1_ratio=l1_ratio,
+                logistic=is_classif)
+            np.testing.assert_almost_equal(alphas.max(), alpha_max)
+            np.testing.assert_almost_equal(n_alphas, len(alphas))
+
+
+def test_space_net_alpha_grid_same_as_sk():
+    try:
+        from sklearn.linear_model.coordinate_descent import _alpha_grid
+        iris = load_iris()
+        X = iris.data
+        y = iris.target
+        np.testing.assert_almost_equal(_space_net_alpha_grid(
+            X, y, n_alphas=5), X.shape[0] * _alpha_grid(X, y, n_alphas=5,
+                                                        fit_intercept=False))
+    except ImportError:
```

Comment:
OK, the feature seems to be asbsent even in 0.13. So I'll simply skip the testcase in the event of ImportError (as it's already done).